crates/core/src/globals.rs

```diff
@@ -172,113 +133,18 @@ mod tests {
 
     use super::inject_javy_globals;
     use anyhow::Result;
-    use std::cell::RefCell;
-    use std::rc::Rc;
-    use std::{cmp, io};
-
-    #[test]
-    fn test_console_log() -> Result<()> {
-        let mut stream = SharedStream::default();
-
-        let runtime = runtime::new_runtime()?;
-        let ctx = runtime.context();
-        inject_javy_globals(&runtime, stream.clone(), stream.clone())?;
-
-        ctx.eval_global("main", "console.log(\"hello world\");")?;
-        assert_eq!(b"hello world\n", stream.buffer.borrow().as_slice());
-
-        stream.clear();
-
-        ctx.eval_global("main", "console.log(\"bonjour\", \"le\", \"monde\")")?;
-        assert_eq!(b"bonjour le monde\n", stream.buffer.borrow().as_slice());
-
-        stream.clear();
-
-        ctx.eval_global(
-            "main",
-            "console.log(2.3, true, { foo: 'bar' }, null, undefined)",
-        )?;
-        assert_eq!(
-            b"2.3 true [object Object] null undefined\n",
-            stream.buffer.borrow().as_slice()
-        );
-        Ok(())
-    }
-
-    #[test]
-    fn test_console_error() -> Result<()> {
-        let mut stream = SharedStream::default();
-
-        let runtime = runtime::new_runtime()?;
-        let ctx = runtime.context();
-        inject_javy_globals(&runtime, stream.clone(), stream.clone())?;
-
-        ctx.eval_global("main", "console.error(\"hello world\");")?;
-        assert_eq!(b"hello world\n", stream.buffer.borrow().as_slice());
-
-        stream.clear();
-
-        ctx.eval_global("main", "console.error(\"bonjour\", \"le\", \"monde\")")?;
-        assert_eq!(b"bonjour le monde\n", stream.buffer.borrow().as_slice());
-
-        stream.clear();
-
-        ctx.eval_global(
-            "main",
-            "console.error(2.3, true, { foo: 'bar' }, null, undefined)",
-        )?;
-        assert_eq!(
-            b"2.3 true [object Object] null undefined\n",
-            stream.buffer.borrow().as_slice()
-        );
-        Ok(())
-    }
 
     #[test]
     fn test_text_encoder_decoder() -> Result<()> {
-        let stream = SharedStream::default();
         let runtime = runtime::new_runtime()?;
         let ctx = runtime.context();
-        inject_javy_globals(&runtime, stream.clone(), stream.clone())?;
+        inject_javy_globals(&runtime)?;
         ctx.eval_global(
             "main",
-            "let encoder = new TextEncoder(); let buffer = encoder.encode('hello'); let decoder = new TextDecoder(); console.log(decoder.decode(buffer));"
+            "let encoder = new TextEncoder(); let buffer = encoder.encode('hello'); let decoder = new TextDecoder(); globalThis.foo = decoder.decode(buffer);"
         )?;
-        assert_eq!(b"hello\n", stream.buffer.borrow().as_slice());
+        assert_eq!("hello", ctx.global_object()?.get_property("foo")?.as_str()?);
```

Comment:
We no longer can pass an in-memory stream into the `console` implementation provided by the `javy-apis` crate so I rewrote this test to write the decoded value to a global variable which we can inspect directly.